src/java/htsjdk/samtools/util/SequenceUtil.java

```diff
@@ -62,9 +86,35 @@ public static String reverseComplement(final String sequenceData) {
         return htsjdk.samtools.util.StringUtil.bytesToString(bases);
     }
 
-    /** Attempts to efficiently compare two bases stored as bytes for equality. */
+    /**
+     * Attempts to efficiently compare two bases stored as bytes for equality.
+     * First does simple comparison if the bases are both in [ACGTN]
+     * Then checks if the bases are any of the IUPAC ambiguous characters and handles matches based on inclusion in the ambiguity set
+     */
     public static boolean basesEqual(byte lhs, byte rhs) {
         if (lhs == rhs) return true;
+
+        // Convert to UPPER case.
+        if (lhs > 90) lhs -= 32;
+        if (rhs > 90) rhs -= 32;
+        if ((isValidBase(lhs, VALID_BASES_WITH_N_UPPER) && (isValidBase(rhs, VALID_BASES_WITH_N_UPPER)))) {
+            // Both lhs and rhs are [A, C, G, T, N] - NOT iupac ambiguity codes
+            // TODO - is this ever called with '.' rather than 'N'??
+            return lhs == rhs;
+        }
+        // If here, one of the two bases is IUPAC ambiguous
+        final Set<Byte> rightSide = iupacBaseMappings.get(rhs);
+        for (byte b : iupacBaseMappings.get(lhs)) {
```

Comment:
Not all sequence is human ;)

I didn't go all the way to creating a PR, but here's a gist that gives an implementation using bitwise comparisons: https://gist.github.com/tfenne/cc49c01e663a7bae4957

@gbggrant I grabbed your expanded tests and ran them using this class, and modifying SequenceUtil.basesEqual() to look like this:

    public static boolean basesEqual(final byte lhs, final byte rhs) {
        if (lhs == rhs) return true;
        else return BaseUtil.basesEqual(lhs, rhs);
    }

All the tests pass.  It is, of course, up to you whether to incorporate this or not.